lib/auth/grpcserver.go

```diff
@@ -2083,15 +2083,35 @@ func maybeDowngradeRoleVersionToV7(role *types.RoleV6, clientVersion *semver.Ver
 	// which can cause panics since it causes a race condition.
 	role = apiutils.CloneProtoMsg(role)
 	role.Version = types.V7
+	role = downgradeSAMLIdPRBAC(role)
 
-	reason := fmt.Sprintf(`Role V8 is only supported from the client version %q and above.`, minSupportedRoleV8Version)
+	reason := fmt.Sprintf(`Role V8 is only supported from the client version %q and above.`+
+		`Access to SAML IdP will be disabled.`, minSupportedRoleV8Version)
 	if role.Metadata.Labels == nil {
 		role.Metadata.Labels = make(map[string]string, 1)
 	}
 	role.Metadata.Labels[types.TeleportDowngradedLabel] = reason
+
 	return role
 }
 
+// downgradeSAMLIdPRBAC disables access to all saml_idp_service_provider
+// resources. The RBAC for saml_idp_service_provider resource before role V8
+// is a blanket allow/deny rule. Since the saml resource can now be
+// scoped per resource labels, disabling access on downgraded role is
+// a simpler and safer option.
+func downgradeSAMLIdPRBAC(downgradedRole *types.RoleV6) *types.RoleV6 {
+	options := downgradedRole.GetOptions()
+	options.IDP = &types.IdPOptions{
+		SAML: &types.IdPSAMLOptions{
+			Enabled: types.NewBoolOption(false),
```

Comment:
The slight alternative we discussed was to check for wildcard allow/deny labels in role v8 and based on that, optionally enable access to the downgraded role. I've opted to not to support that for now as the client mismatch is a transient issue during server upgrades and keeping it simple and disallowing access is a safer bet. That will urge users to upgrade their client at minimum. Open to any suggestion otherwise.